projects/cashmere/src/lib/pipes/highlight/highlight-pipe.md

```diff
@@ -0,0 +1,50 @@
+The Highlight pipe is used to identify any text from the user's query.
+
+#### API
+
+```ts
+transform(value: string): string
+```
+
+`value`: the `string` value to query
+
+#### Importing the Highlight Pipe
+
+To use the `HighlightPipe`, import the `HighlightPipeModule` into your app's `CashmereModule` (which in turn is imported into your app):
+
+```ts
+import {NgModule} from '@angular/core';
+import {HighlightPipeModule} from '@healthcatalyst/cashmere';
+
+@NgModule({
+    exports: [HighlightPipeModule]
+})
+export class CashmereModule {}
+```
+
+Alternatively, each pipe in Cashmere is also exported from `PipesModule`, if you wish to import all pipes from the `@healthcatalyst/cashmere` package:
+
+```ts
+import {NgModule} from '@angular/core';
+import {PipesModule} from '@healthcatalyst/cashmere';
+
+@NgModule({
+    exports: [PipesModule]
+})
+export class CashmereModule {}
+```
+
+#### Using the Highlight Pipe in character mode
+
+Pipe a `string` value to the Highlight pipe representing query. If the query is found in the text, the value is then highlighted disregarding case sensitivity.
+
+```html
+{{ 'at' }}
+<!-- will highlight At…at…  -->
+{{ 'At' }}
+<!-- will highlight At…at…  -->
+```
+
+#### Edge cases
```

Comment:
Just looking at the unit tests Joe built, we probably want to edit these edge cases.  For example, what happens if the text contains HTML, what happens when the search term is blank, etc.

If you want to take a crack at this @Victor1818 - you can use the language @joeskeen built into his unit tests directly.  For example, one edge could be taken from line 70 in the spec file, and you could add it as a bullet in these edge cases like: "When the text value for the Highlight Pipe contains HTML, it should highlight matches on the escaped HTML".  And you could do something similar for just about all his tests in the spec - just in bullet list form under this Edge cases section